RELEASE.md

```diff
@@ -0,0 +1,28 @@
+# Release Process
+
+This document describes the process for releasing a new version of the `actors-utils` project.
+
+## Current State
+
+1. On a release pull request creation, a [Release Checker](.github/workflows/release-check.yml) workflow will run. It will perform the following actions:
+    1. Extract the version from the modified `Cargo.toml` files. Process each crate in the workspace **independently**.
+    2. Check if a git tag for the version, using the `crate_name@version` as the pattern, already exists. Continue only if it does not.
+    3. Create a draft GitHub release with the version as the tag.
+    4. Comment on the pull request with a link to the draft release.
+2. **[MANUAL]** Run `cargo publish --dry-run` for each crate that is proposed to be released in the reverse dependency order.
```

Comment:
How does someone find the reverse dependency order?

Reply:
And how does this get done for the crate? we have a workspace here so do we need to `-p` to do this or do we `cd` into the directory to run this? I'm not clear on how publishing on workspaces works so this is a bit of a mystery to me.